CMakeLists.txt

```diff
@@ -134,32 +134,29 @@ ENDIF()
 # Configure C++ standard if not externally specified (will actually be
 # set after CppUnit check below). Use the variable CMAKE_CXX_STANDARD
 # since it will actually be used for this purposes starting in CMake 3.1.
+SET(GR_C_STANDARD 11)
+SET(GR_CXX_STANDARD 11)
 
 IF(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-    SET(CMAKE_CXX_STANDARD 11)
+    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++${GR_CXX_STANDARD}")
 ELSEIF(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
-    SET(CMAKE_CXX_STANDARD 11)
+    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++${GR_CXX_STANDARD}")
 ELSEIF(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
-    SET(CMAKE_CXX_STANDARD 11)
+    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++${GR_CXX_STANDARD}")
```

Comment:
I'll really need an explanation why setting this ourselves is better than letting CMake handle this? I know you're not doing it for the fun of it, but at this point, my understanding of the problem just doesn't explain why *not* setting the `CMAKE_CX*_STANDARD` is better here.

Reply:
see the discussion with @mbr0wn about cmake setting `gnu$version` flags, which is not really helpful.